src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/HMACMD5.cs

```diff
@@ -29,7 +29,7 @@ public HMACMD5(byte[] key)
             HashSizeValue = _hMacCommon.HashSizeInBits;
         }
 
-        public override byte[] Key
+        public override byte[]? Key
```

Comment:
I don't think any of the HMAC types can ever have a null key.  I assume that they're declared nullable because of their base class; but we can probably just restrict these.  (At least the concrete types, HMACMD5, HMACSHA1, HMACSHA256, HMACSHA384, HMACSHA512)

Reply:
Yes because of base, but now I see that base.Key always set with constructor

Reply:
@bartonjs, does the base class permit null to be set just because we neglected to previously check for null and now it would be a breaking change?  If yes, it would seem to make sense to say that the intent is that null is allowed.  Or does the base class permit null to be set because it's actually meaningful to do so / we've documented it as something that's useful?  If yes, it's an unfortunate difference between the base and derived classes, and we should leave the base as having a nullable setter, make the derived have a non-nullable setter, and suppress the compiler warning.

Reply:
`KeyedHashAlgorithm` (the defining type) is abstract, and requires a derived type to set a protected field to exit the null state.  None of the non-abstract derived types ever allow it to be null, though.

So I think that would mean we change (in S.S.C.Primitives) `KeyedHashAlgorithm.Key` to be non-null, and `KeyedHashAlgorithm.KeyValue` to be non-null, initialized to `null!`?

Reply:
OK, then I need update the other API first and update this one after that change merged